sympy/physics/control/routh_table.py

```diff
@@ -51,64 +54,106 @@ def negative_real_part_conditions(polynomial, var, /, *, domain=None):
     """
     p: Poly = Poly(polynomial, var, domain = domain)
 
-    _, p = p.clear_denoms(convert=True)
+    try:
+        return p.routh_hurwitz()
 
-    coeffs = p.all_coeffs()
+    except NotImplementedError:
+        # If the algorithm is non implemented for a specific domain,
+        # we calculate it with expressions
+        coeffs = p.all_coeffs()
```

Comment:
You can include the expression implementation with the others in polys. It is better not to have a separate implementation here. The EXRAW domain is precisely just "expressions" so there can be an implementation for EXRAW.